great_expectations/checkpoint/checkpoint.py

```diff
@@ -102,7 +102,7 @@ class BaseCheckpoint(ConfigPeer):
     def __init__(
         self,
         checkpoint_config: CheckpointConfig,
-        data_context: AbstractDataContext,
+        data_context: AbstractDataContext | None = None,
```

Comment:
I'd like to fully decouple this requirement but for now, let's make it optional and have the factory supply it on `add`